bigquery/google/cloud/bigquery/table.py

```diff
@@ -684,6 +684,29 @@ def fetch_data(self, max_results=None, page_token=None, client=None):
         iterator._NEXT_TOKEN = 'pageToken'
         return iterator
 
+    def row_from_mapping(self, mapping):
+        """Convert a mapping to a row tuple using the schema.
+
+        :type mapping: dict
+        :param mapping: Mapping of row data.
+
+        :rtype: tuple
+        :returns: Tuple matching the table's schema
+        :raises: ValueError if table's schema is not set
+        """
+        if len(self._schema) == 0:
+            raise ValueError(_TABLE_HAS_NO_SCHEMA)
+
+        row = []
+        for field in self.schema:
+            if field.mode == 'REQUIRED':
+                row.append(mapping[field.name])
+            elif field.mode == 'REPEATED':
+                row.append(mapping.get(field.name, ()))
+            else:  # NULLABLE
```

Comment:
Could make an explicit `field.mode == 'NULLABLE'` and then have a 4th branch that just barfs?